core/src/main/java/org/mskcc/cbio/oncokb/util/MainUtils.java

```diff
@@ -860,4 +861,44 @@ public static Integer compareAnnotationSearchQueryType(AnnotationSearchQueryType
             return asc ? -1 : 1;
         return (PRIORITIZED_QUERY_TYPES.indexOf(q1) - PRIORITIZED_QUERY_TYPES.indexOf(q2)) * (asc ? 1 : -1);
     }
+
+    public static int findDigitEndIndex(String str, int startIndex) {
+        int i = startIndex;
+        while (i < str.length() && Character.isDigit(str.charAt(i))) {
+            i++;
+        }
+        return i;
+    }
+
+    public static GenomicLocation parseChromosomeAndRangeFromHGVSg(String hgvsg) {
```

Comment:
Why are we copying the method over?

Reply:
The original implementation used the NotationConverter to parse HGVSg and the regex was the main bottleneck. Also, the `genomicLocationShouldBeAnnotated` only used the chromosome and ranges, so we don't need to parse the entire HGVSg. I believe the parseHGVSg was about 7% of the running state.

Reply:
I can revert to using NotationConverter if this is not that big of an issue.

Reply:
Can we add some test cases here?